### ISSUE TITLE
Espace fine dans les URLs

### ISSUE THREAD

**Issue by vslonska** [victorloux/espaceFine#5]
Bonjour,

Merci pour cette fonction. Seul un détail que je remarque : si on écrit une url, sans faire un lien, la fonction place une espace entre le le `https` et le `:`.

On obtient donc `https ://github.com/victorloux/espaceFine/issues/new`.

Bonne journée,

**Comment by victorloux** [comment]
Merci d'avoir remarqué ceci ! Je vais faire quelques essais pour voir s'il est possible d'exclure le pattern `://` sans avoir trop d'impact sur la performance.

De façon générale cela peut être un problème pour d'autres situations (émoticônes, sections dans d'autres langues…) ; une solution alternative serait de n'activer le script que si une espace normale existe déjà, plutôt que de toujours en insérer une même si elle est manquante.

**Comment by vslonska** [comment]
Oui, ça peut être une bonne idée. Je crois que c'est ce que font les scripts qui gèrent ça dans Spip par exemple.

De rien, quoi qu'il en soit, et merci.

**Comment by anthonymasure** [comment]
Je pense aussi que c'est mieux de n'exécuter le script qu'en cas d'espaces simples, par exemple dans le cas où du texte FR voisine avec du EN

**Comment by thom4parisot** [comment]
J'ai rencontré le problème dans un autre contexte. J'ai modifié la regex de…

```
/(\u00AB|\u2014)(?:\s+)?|(?:\s+)?([\?!:;\u00BB])/g
```

… à … 

```
 /(\u00AB|\u2014)(?:\s+)|(?:\s+)([?!:;\u00BB])/g
```

**Comment by victorloux** [comment]
Merci @oncletom, je vais modifier ça + les exemples quand j'aurais un peu de temps.

Pour la langue je pense qu'un meilleur heuristique serait aussi de n'appliquer le script sur le sélecteur `:lang(fr)`, ainsi tout passage dans une autre langue (correctement marqué avec `lang="en"`) ne serait pas incorrectement modifié, même s'il y a des espaces.
